libs/s25main/EconomyModeHandler.cpp

```diff
@@ -33,56 +34,29 @@
 
 EconomyModeHandler::EconomyModeHandler(unsigned endFrame) : endFrame(endFrame), gfLastUpdated(0)
 {
-    const std::vector<GoodType> specialGoodPool({
-      GD_TONGS,
-      GD_HAMMER,
-      GD_AXE,
-      GD_SAW,
-      GD_PICKAXE,
-      GD_SHOVEL,
-      GD_CRUCIBLE,
-      GD_RODANDLINE,
-      GD_SCYTHE,
-      GD_CLEAVER,
-      GD_ROLLINGPIN,
-      GD_BOW,
-    });
-
-    const std::vector<GoodType> commonGoodPool({
-      GD_BEER,
-      GD_WATER,
-      GD_BOAT,
-      GD_SWORD,
-      GD_IRON,
-      GD_FLOUR,
-      GD_FISH,
-      GD_BREAD,
-      GD_WOOD,
-      GD_BOARDS,
-      GD_STONES,
-      GD_GRAIN,
-      GD_COINS,
-      GD_GOLD,
-      GD_IRONORE,
-      GD_COAL,
-      GD_MEAT,
-      GD_HAM,
-    });
+    const auto specialGoodPool =
```

Comment:
You can use constexpr here so the array is created at compiletime. Also the template argument to make_array shouldn't be required. It is optional in case mixed values are used

Reply:
I would argue that while the template argument is not necessary it's actually useful here so we get a type check for the elements put inside. But you seem to care much more about these details than me, so I removed the template argument.